generic/hwinfo.py

```diff
@@ -0,0 +1,134 @@
+            self.is_fail += 1
+        return
+
+    def setUp(self):
+        dist = distro.detect()
+        architecture = os.uname()[4]
+        if "ppc" not in architecture:
+            self.skip("supported only on Power platform")
+        if dist.name == 'redhat':
+            self.skip('Hwinfo not supported on RHEL')
+        sm = SoftwareManager()
+        if not sm.check_installed("hwinfo") and not sm.install("hwinfo"):
+            self.error("Fail to install hwinfo required for this test.")
+
+    def test(self):
+        self.log.info("===============Executing hwinfo tool test===============")
+        self.run_cmd("hwinfo --all")
+        self.run_cmd("hwinfo --arch")
+        self.run_cmd("hwinfo --bios")
+        self.run_cmd("hwinfo --block")
+        self.run_cmd("hwinfo --bluetooth")
+        self.run_cmd("hwinfo --braille")
+        self.run_cmd("hwinfo --bridge")
+        self.run_cmd("hwinfo --camera")
+        self.run_cmd("hwinfo --cdrom")
+        self.run_cmd("hwinfo --chipcard")
+        self.run_cmd("hwinfo --cpu")
+        self.run_cmd("hwinfo --disk")
+        self.run_cmd("hwinfo --dsl")
+        self.run_cmd("hwinfo --dvb")
+        self.run_cmd("hwinfo --fingerprint")
+        self.run_cmd("hwinfo --floppy")
+        self.run_cmd("hwinfo --framebuffer")
+        self.run_cmd("hwinfo --gfxcard")
+        self.run_cmd("hwinfo --hub")
+        self.run_cmd("hwinfo --ide")
+        self.run_cmd("hwinfo --isapnp")
+        self.run_cmd("hwinfo --isdn")
+        self.run_cmd("hwinfo --joystick")
+        self.run_cmd("hwinfo --keyboard")
+        self.run_cmd("hwinfo --memory")
+        self.run_cmd("hwinfo --modem")
+        self.run_cmd("hwinfo --monitor")
+        self.run_cmd("hwinfo --mouse")
+        self.run_cmd("hwinfo --netcard")
+        self.run_cmd("hwinfo --network")
+        self.run_cmd("hwinfo --partition")
+        self.run_cmd("hwinfo --pci")
+        self.run_cmd("hwinfo --pcmcia")
+        self.run_cmd("hwinfo --pcmcia-ctrl")
+        self.run_cmd("hwinfo --pppoe")
+        self.run_cmd("hwinfo --printer")
+        self.run_cmd("hwinfo --redasd")
+        self.run_cmd("hwinfo --reallyall")
+        self.run_cmd("hwinfo --scanner")
+        self.run_cmd("hwinfo --scsi")
+        self.run_cmd("hwinfo --smp")
+        self.run_cmd("hwinfo --sound")
+        self.run_cmd("hwinfo --storage-ctrl")
+        self.run_cmd("hwinfo --sys")
+        self.run_cmd("hwinfo --tape")
+        self.run_cmd("hwinfo --tv")
+        self.run_cmd("hwinfo --uml")
+        self.run_cmd("hwinfo --usb")
+        self.run_cmd("hwinfo --usb-ctrl")
+        self.run_cmd("hwinfo --vbe")
+        self.run_cmd("hwinfo --wlan")
+        self.run_cmd("hwinfo --xen")
+        self.run_cmd("hwinfo --zip")
+        self.run_cmd("hwinfo --short")
+        self.run_cmd("hwinfo --listmd")
+        disk_name = process.system_output("df -h | egrep '(s|v)d[a-z][1-8]' | "
+                                          "tail -1 | cut -d' ' -f1",
```

Comment:
above command has extra double qoute, which may fail.
Why don't we use 'disk' module in avocado utils to get the disk node ? Or
Pass the disk from yaml file, which is always good.  self.disk = self.params.get('disk')